linode/helper/planmodifiers/caseinsensitiveset.go

```diff
@@ -0,0 +1,58 @@
+package planmodifiers
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// CaseInsensitiveSet ensures that a set does not trigger
+// diffs on planned values with different cases.
+//
+// NOTE: This is not implemented as custom type because custom type semantic equality
+// checks do not have granular control over diffs.
+func CaseInsensitiveSet() planmodifier.Set {
+	return &caseInsensitiveSetPlanModifier{}
+}
+
+type caseInsensitiveSetPlanModifier struct{}
+
+func (d *caseInsensitiveSetPlanModifier) Description(ctx context.Context) string {
+	return "Ensures a set does not trigger diffs on planned values with different cases."
+}
+
+func (d *caseInsensitiveSetPlanModifier) MarkdownDescription(ctx context.Context) string {
+	return d.Description(ctx)
+}
+
+func (d *caseInsensitiveSetPlanModifier) PlanModifySet(ctx context.Context, req planmodifier.SetRequest, resp *planmodifier.SetResponse) {
+	oldEntryMap := make(map[string]string)
+	resultList := make([]string, 0)
```

Comment:
```suggestion
resultList := make([]string)
```
We can ignore the `0` to make it simpler

Reply:
Never mind, my memory was incorrect...